pkg/controller/disruption/disruption.go

```diff
@@ -452,9 +487,15 @@ func (dc *DisruptionController) trySync(pdb *policy.PodDisruptionBudget) error {
 		return err
 	}
 
-	currentHealthy := countHealthyPods(pods)
-	err = dc.updatePdbSpec(pdb, currentHealthy, desiredHealthy, expectedCount)
+	currentTime := time.Now()
+	disruptedPods, recheckTime := buildDisruptedPodMap(pods, pdb, currentTime)
+	currentHealthy := countHealthyPods(pods, disruptedPods, currentTime)
+	err = dc.updatePdbStatus(pdb, currentHealthy, desiredHealthy, expectedCount, disruptedPods)
 
+	if err == nil && recheckTime != nil {
+		// Delaying interface stores only one value with the lowest timestamp.
```

Comment:
Well, please update the comment to make it clearer, as what it says now is confusing. How about "there is always at most one PDB with a particular name in the queue, and each PDB in the queue is associated with the lowest timestamp that was supplied when a PDB with that name was added"?